tests/lib/mountinfo-tool

```diff
@@ -0,0 +1,379 @@
+            n = len(seen)
+            seen[key] = n
+            return n
+
+    def fn(m: Match[str]) -> str:
+        return 'loop{}'.format(alloc_n(int(m[1])))
+    entry.mount_source = re.sub('loop(\\d+)', fn, entry.mount_source)
+
+
+def renumber_mount_ids(entry: MountInfoEntry, seen: Dict[int, int]) -> None:
+    """renumber_mount_ids re-numbers mount and parent mount IDs."""
+    def alloc_n(mount_id: int) -> int:
+        key = mount_id
+        try:
+            return seen[key]
+        except KeyError:
+            n = len(seen)
+            seen[key] = n
+            return n
+    # NOTE: renumber the parent ahead of the mount to get more
+    # expected relationship between them.
+    entry.parent_id = alloc_n(entry.parent_id)
+    entry.mount_id = alloc_n(entry.mount_id)
+
+
+def renumber_devices(
+        entry: MountInfoEntry, seen: Dict[Device, Device]) -> None:
+    """renumber_devices re-numbers major:minor device numbers."""
+    def alloc_n(dev: Device) -> Device:
+        key = dev
+        try:
+            return seen[key]
+        except KeyError:
+            # We haven't seen the major:minor pair precisely but perhaps we've
+            # seen the major number already? Check if this major is already
+            # remapped, if so reuse that value. If not just allocate the next
+            # one based on cardinality of the set of major numbers we've seen.
+            major: int
+            for orig, remapped in seen.items():
+                if orig.major == dev.major:
+                    major = remapped.major
+                    break
+            else:
+                major = len({orig.major for orig in seen})
+            # Allocate the next minor number based on the cardinality of the
+            # set of minor numbers matching the major number.
+            minor = len({orig.minor for orig in seen
+                        if orig.major == dev.major})
+            n = Device.pack(major, minor)
+            seen[key] = n
+            return n
+    entry.dev = alloc_n(entry.dev)
+
+
+def rewrite_renumber(entries: List[MountInfoEntry]) -> None:
+    """rewrite_renumber applies all re-numbering helpers."""
+    seen_opt_fields: Dict[int, int] = {}
+    seen_loops: Dict[int, int] = {}
+    seen_snap_revs: Dict[Tuple[str, str], int] = {}
+    seen_mount_ids: Dict[int, int] = {}
+    seen_devices: Dict[Device, Device] = {}
+    for entry in entries:
+        renumber_mount_ids(entry, seen_mount_ids)
+        renumber_devices(entry, seen_devices)
+        renumber_snap_revision(entry, seen_snap_revs)
+        renumber_opt_fields(entry, seen_opt_fields)
+        renumber_loop_devices(entry, seen_loops)
+
+
+def rewrite_rename(entries: List[MountInfoEntry]) -> None:
+    """rewrite_renameapplies all re-naming helpers."""
+    # TODO: allocate devices like everything else above.
+    for entry in entries:
+        entry.mount_source = re.sub(
+            '/dev/[sv]d([a-z])', '/dev/sd\\1', entry.mount_source)
```

Comment:
Should be good enough for spread tests and VMs

Reply:
Yeah, it's written based on experience :)